### PR TITLE
Update samples and readme for  Content Safety sdk

### PR DESCRIPTION
**Pull request by mengaims**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

**Comment by mengaims** [comment]
@heaths I've added the readme and refine samples for Content Safety in this PR, could you help review?

File: sdk/contentsafety/Azure.AI.ContentSafety/CHANGELOG.md
Comment (heaths) [inline sdk/contentsafety/Azure.AI.ContentSafety/CHANGELOG.md:3]: Did you just not hit this date? In general, we leave this as "Unreleased" until right before we ship.

File: sdk/contentsafety/Azure.AI.ContentSafety/README.md
Comment (heaths) [inline sdk/contentsafety/Azure.AI.ContentSafety/README.md:6]:
```suggestion
* Text Blocklist Management APIs: The default AI classifiers are sufficient for most content safety needs; however, you might need to screen for terms that are specific to your use case. You can create blocklists of terms to use with the Text API.
```

File: sdk/contentsafety/Azure.AI.ContentSafety/README.md
Comment (heaths) [inline sdk/contentsafety/Azure.AI.ContentSafety/README.md:8]: You should point to your specific nuget package (or what it will be once published). All other links need to be specific to your SDK / service. These are all deterministic, so you could author them now. It's important to do that before you GA because:

1. The README.md is packaged in the nupkg.org, so if you fix it later in the repo it won't be reflected in your package and will be wrong.
2. We tag the repo when a package is released e.g. Azure.AI.ContentSafety_1.0.0 and use that link in several places rather than just point to HEAD for main, so the README.md would be wrong in that case as well.

File: sdk/contentsafety/Azure.AI.ContentSafety/README.md
Comment (heaths) [inline sdk/contentsafety/Azure.AI.ContentSafety/README.md:59]: I recommend putting space between paragraphs, tables, headings, lists (not items), etc. If you install the Microsoft Learn Pack, it has a linter that will help format this.

While the README.md does get rendered in many cases, we want it to be readable even as plain text - one of the goals of markdown.

File: sdk/contentsafety/Azure.AI.ContentSafety/README.md
Comment (heaths) [inline sdk/contentsafety/Azure.AI.ContentSafety/README.md:151]: This is awfully verbose. I suggest the null coalescence operator:
```suggestion
Console.WriteLine("Hate severity: {0}", response.Value.HateResult?.Severity ?? "0.0");
Console.WriteLine("SelfHarm severity: {0}", response.Value.SelfHarmResult?.Severity ?? "0.0");
Console.WriteLine("Sexual severity: {0}", response.Value.SexualResult?.Severity ?? "0.0");
Console.WriteLine("Violence severity: {0}", response.Value.ViolenceResult?.Severity ?? "0.0");
```
`Console.WriteLine` also implements string formatting more efficiently, so you're wasting time and showing atypical code so don't use `string.format`.

File: sdk/contentsafety/Azure.AI.ContentSafety/README.md
Comment (heaths) [inline sdk/contentsafety/Azure.AI.ContentSafety/README.md:132]: Don't pass `String.Format` to `Console.WriteLine`, which also supports formattables e.g., `Console.WriteLine(string format, params object[] args)`.
```suggestion
    Console.WriteLine("Analyze text failed.\nStatus code: {0}, Error code: {1}, Error message: {2}", ex.Status, ex.ErrorCode, ex.Message);
```

**Review by heaths — CHANGES_REQUESTED** [review]

**Comment by pallavit** [comment]
Can we please include the SDK in the codeowners file - https://github.com/Azure/azure-sdk-for-net/blob/main/.github/CODEOWNERS

File: sdk/contentsafety/Azure.AI.ContentSafety/CHANGELOG.md
Comment (mengaims) [reply sdk/contentsafety/Azure.AI.ContentSafety/CHANGELOG.md:3]: Yes we missed the 0522 release date, and decided to release by 06/06. Sure I will keep "Unreleased" in following versions. Since the date is very near now, can we leave it "2023-06-06" for now?

File: sdk/contentsafety/Azure.AI.ContentSafety/README.md
Comment (mengaims) [reply sdk/contentsafety/Azure.AI.ContentSafety/README.md]: Fixed.

File: sdk/contentsafety/Azure.AI.ContentSafety/README.md
Comment (mengaims) [reply sdk/contentsafety/Azure.AI.ContentSafety/README.md]: Fixed.

File: sdk/contentsafety/Azure.AI.ContentSafety/README.md
Comment (mengaims) [reply sdk/contentsafety/Azure.AI.ContentSafety/README.md]: Fixed

File: sdk/contentsafety/Azure.AI.ContentSafety/README.md
Comment (mengaims) [reply sdk/contentsafety/Azure.AI.ContentSafety/README.md]: Thanks for the suggestion! It's very helpful. Reformat the readme by linter.

File: sdk/contentsafety/Azure.AI.ContentSafety/README.md
Comment (mengaims) [reply sdk/contentsafety/Azure.AI.ContentSafety/README.md]: Thanks for the elaboration! Changed to product specific links.

**Comment by mengaims** [comment]
@heaths I've reply all the comments, please help check, thanks!

**Review by heaths — CHANGES_REQUESTED** [review]
You need to run _eng/scripts/CodeChecks.ps1 contentsafety_ to make sure everything is updated.

Actually, I see now why the build didn't fail like it should've: you need to add a package-level "samples" with markdown files that reference the snippets. See https://github.com/Azure/azure-sdk-for-net/blob/main/samples/README.MD.

**Comment by mengaims** [comment]
> You need to run _eng/scripts/CodeChecks.ps1 contentsafety_ to make sure everything is updated.
> 
> Actually, I see now why the build didn't fail like it should've: you need to add a package-level "samples" with markdown files that reference the snippets. See https://github.com/Azure/azure-sdk-for-net/blob/main/samples/README.MD.

Hi @heaths , I've  added a package-level "samples", please check.
And I've also ran _eng/scripts/CodeChecks.ps1 contentsafety_ locally, it's wired that it always report following error:
```
error : Missing project. Solution references a project which does not exist: D:\src\mengaims\azure-sdk-for-net\sdk\contentsafety\Azure.AI.ContentSafety\Θí╣τ¢«. [D:\src\mengaims\azure-sdk-for-net\sdk\contentsafety\Azure.AI.ContentSafety\Azure.AI.ContentSafety.sln]
error : Missing project. Solution references a project which does not exist: D:\src\mengaims\azure-sdk-for-net\sdk\contentsafety\Azure.AI.ContentSafety\--. [D:\src\mengaims\azure-sdk-for-net\sdk\contentsafety\Azure.AI.ContentSafety\Azure.AI.ContentSafety.sln]
```
Is it the same case from your side? I'm confused with these errors, since I've double checked there is no invalid project reference in this solution file, and the CI build all look good. Besides I tried to search for some information but did not find the description of this script in https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md.
So that I'd like to ignore them, are you ok with it? Could you help approve if no other concern?

**Review by heaths — APPROVED** [review]

**Pull request merged**
